staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -130,34 +144,35 @@ func EachListItem(obj runtime.Object, fn func(runtime.Object) error) error {
 	if len == 0 {
 		return nil
 	}
-	takeAddr := false
-	if elemType := items.Type().Elem(); elemType.Kind() != reflect.Pointer && elemType.Kind() != reflect.Interface {
-		if !items.Index(0).CanAddr() {
-			return fmt.Errorf("unable to take address of items in %T for EachListItem", obj)
-		}
-		takeAddr = true
-	}
```

Comment:
it looks like this logic (and the `if takeAddr { raw = raw.Addr() }` was dropped ... can we track back why it was added and if there are callers that could be broken by dropping this?

Reply:
by not taking the `Addr()` would a list of `[]runtime.RawExtension` that previously matched the `*runtime.RawExtension` case no longer match?

Reply:
The reason for the addition is hard to trace. The takeAddr code was added on 2017/7/20. . commit-message is "start the apimachinery repo" 😢